### DISCUSSION TITLE
DigitalOcean App: pscale connect: database x and branch y does not exist in org z

### DISCUSSION THREAD

**Discussion by ppalmeida** [planetscale/discussion#166] [Q&A] [answered]
I have a DigitalOcean App (PHP / Laravel) and I'm trying to use PlanetScale. I saw in PlanetScale docs that I shouldn't be using service token to connect to a prod database. But, for now, to make it work once, that would be enough. I was following [this answer in the DigitalOcean community, trying to make my app work with PS](https://www.digitalocean.com/community/questions/how-can-we-configure-planetscale-with-app-platform) . I created the org, the database, and the service token in PlanetScale. I can use `pscale` CLI to connect to my database from my computer (OSX) using `pscale auth login` (opens a browser, etc), then `pscale connect ....`. It works fine and I can run queries on that, like `show databases` and `show tables from mydatabase`.

The issue is trying to use it in DigitalOcean: when I try to follow that answer in DigitalOcean App, it throws an error during the build process:

> Error: database `my_db_name` and branch `main` does not exist in organization `my_org_name`

Of course, in DigitalOcean I can not use `pscale auth login` during the script initialization. So, I used the scripts suggested by the solution link above. In summary:

The Build Script:  (download and give run permission to pscale tool):

> export PSCALE_VERSION="$(curl https://github.com/planetscale/cli/releases/latest -s -L -I -o /dev/null -w '%{url_effective}' | awk '{n=split($1,A,"/v"); print A[n]}')" && \
curl -sL https://github.com/planetscale/cli/releases/download/v$PSCALE_VERSION/pscale_${PSCALE_VERSION}_linux_amd64.tar.gz | tar -xz -C ./ && \
chmod +x ./pscale

Then in Run Script: (connect to PlanetScale and start apache2 in Laravel's `public/` folder):

> ./pscale connect <my database name> <branch main> --execute "heroku-php-apache2 public/"

I put both scripts above in the "Settings" for the DigitalOcean App, as "Build Script" and "Run Script", exactly like was suggested in the answer above.

Then, when I save and Digital Ocean auto-deploys my App, I can see the build breaking saying that the database/branch doesn't exist in my organization. Can this error be a Network error? Does anyone had ever configured Laravel + DigitalOcean App + PlanetScale or has a better tutorial/instructions?

I really want to use PlanetScale and escape from DigitalOcean database services for this project.

**Comment by jgreet** [comment, marked as answer]
:wave: @ppalmeida

It looks like [that DO solution](https://www.digitalocean.com/community/questions/how-can-we-configure-planetscale-with-app-platform?comment=168722) was written right before we made [connection strings](https://docs.planetscale.com/concepts/connection-strings) available.

At this point we would definitely recommend using a connection string over a `pscale connect` proxy in production. We even provide an example Laravel configuration right there in the web app when you create a new password.

We also have a [Laravel tutorial on our doc site](https://docs.planetscale.com/tutorials/connect-laravel-app).

As for why your current setup isn't working, the first thing I'd check is that your service token has `connect_production_branch` access to the database in question:

`pscale service-token show-access <ID>` or take a look in the [web interface](https://docs.planetscale.com/concepts/service-tokens#manage-database-access-permissions).

**Reply by ppalmeida** [reply]
Excelent! Brilliant!

I'm using the connection strings and everything is working fine. For those who's struggling, I put all details of the connection as Laravel's env vars and added `MYSQL_ATTR_SSL_CA` = `/etc/ssl/certs/ca-certificates.crt` and it worked like a charm.

Thank you, @jgreet
